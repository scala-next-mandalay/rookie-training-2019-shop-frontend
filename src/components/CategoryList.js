import React from 'react'
import PropTypes from 'prop-types'
import { Box, List, ListItem, ListItemText } from '@material-ui/core'
import './style.css'

const CategoryList = ({ categories, setCategoryId, handleDrawerClose }) => {
  return (
    <List>
        {categories.map((obj) => (
          <ListItem button key={obj.id} onClick={()=>{
           handleDrawerClose()
            setCategoryId(obj.id)
          }}  >
            <ListItemText>
            <Box fontWeight={600} color='#9A7B66' fontStyle='italic' class='ho'>
              {obj.name}
            </Box>
            </ListItemText>
          </ListItem>
        ))}
      </List>
  )
}

CategoryList.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired).isRequired,
  setCategoryId: PropTypes.func,
  handleDrawerClose: PropTypes.func,
}

export default CategoryList
