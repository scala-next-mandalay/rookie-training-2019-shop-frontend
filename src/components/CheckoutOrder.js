import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { BASEURL_ITEM_IMAGES } from '../constants';
import { Grid, Box, CardMedia, Paper, Divider, Hidden,Link } from '@material-ui/core';
import QuantitySelect from '../containers/QuantitySelect';
import DeleteOutlinedIcon from '@material-ui/icons/DeleteSharp';
import Tooltip from '@material-ui/core/Tooltip';
 import { FormattedMessage } from 'react-intl';


const useStyles = makeStyles(theme => ({
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  descImagePC: {
    width: 70,
    height: 100
  },
  menu: {
    width: 200,
  },
  btnRemoveItem:{
    cursor:'pointer'
  },
  iconSpace:{
     marginTop: theme.spacing(1),
     marginRight: theme.spacing(1),
  }
}));

const CheckoutOrder = ({ cart, totalPrice,totalQuantity,deleteCartItem}) => {
  const removeCartItem= (id)=>{
    deleteCartItem(id);
  };
  const classes = useStyles();
  const mobileView = [];
  const pcView = [];
  const Netprice = () => (
    <Box display="flex" mt={2} pl={3} pb={3}>
      <Grid item xs={false} sm={4}>
        <Box> </Box>
      </Grid>
      <Grid item xs={6} sm={4}>
        <Box><FormattedMessage id="Label.TotalQuantity" defualtMessage="Total Quantity" /></Box>
        <Box><FormattedMessage id="Label.SubTotal" defualtMessage="SubTotal" /> </Box>
        <Box><FormattedMessage id="Label.Tax" defualtMessage="Sales Tax" /></Box>
        <Box><FormattedMessage id="Label.TotalPrice" defualtMessage="Total Price" /></Box>
      </Grid>
      <Grid item xs={6} sm={4} className="text1">
        <Box textAlign="right" pr={3}>{totalQuantity} (Items)</Box>
        <Box textAlign="right" pr={3}> {totalPrice} MMK </Box>
        <Box textAlign="right" pr={3}> 0.0 MMK </Box>
        <Box textAlign="right" pr={3}> {totalPrice} MMK</Box>
      </Grid>
    </Box>
  );
  cart.forEach((cartList, k) => {
    pcView.push(
      <Grid container key={cartList.id} className="text2">
        <Box flexGrow={1} display="flex">
          <Grid item xs={3}>
            <Box textAlign="left" my={2} pl={3}>
              <CardMedia
                className={classes.descImagePC}
                image={BASEURL_ITEM_IMAGES + cartList.image}
                title={cartList.name}
              />
            </Box>
          </Grid>
          <Grid item xs={3}>
            <Box textAlign="left" my={2} fontWeight="100" className="nameTxt">
              {cartList.name}
            </Box>
          </Grid>
          <Grid item xs={2}>
            <Box textAlign="left" my={2} >
              <QuantitySelect id={cartList.id} quantity={cartList.quantity} />
                        </Box>
          </Grid>
        </Box>
        <Grid item xs={2}>
          <Box textAlign="right" my={2} mr={3}>
            <Box pb={2} mr={0}>({cartList.quantity}x{cartList.price})MMK</Box>
            <Box mr={0}> {cartList.subTotal}MMK </Box>
          <Box>  
           <Link className={classes.btnRemoveItem} onClick={()=> removeCartItem(cartList.id)} color="secondary" fontSize="10px">
           <Box mr={0}>
           <Tooltip title="Delete">
           <DeleteOutlinedIcon/>
           </Tooltip>
           </Box>
           </Link>
          </Box>
          </Box>
        </Grid>
        <Grid item xs={12}><Divider /></Grid>
      </Grid>
    );
  });
  cart.forEach((cartList, k) => {
    mobileView.push(
      <Grid container key={cartList.id}>
        <Box flexGrow={1} display="flex">
          <Grid item xs={6}>
            <Box textAlign="left" my={2} pl={3}>
              <CardMedia
                className={classes.descImagePC}
                image={BASEURL_ITEM_IMAGES + cartList.image}
                title={cartList.name}
              />
            </Box>
          </Grid>
          <Grid item xs={6} className="text2">
            <Box textAlign="right" my={2} pr={3} fontWeight="fontWeightBold">
              {cartList.name}
            </Box>
             <Box textAlign="right" my={2} pr={3}>
             <QuantitySelect id={cartList.id} quantity={cartList.quantity} />
             </Box>
            <Box textAlign="right" my={2} pr={3} >
              {cartList.quantity}x {cartList.price}MMK
            </Box>
            <Box textAlign="right" my={2} pr={3}>
              {cartList.subTotal} MMK
            </Box>
            <Box textAlign="right" my={2} pr={3}>
              <Link className={classes.btnRemoveItem} onClick={()=> removeCartItem(cartList.id)} color="secondary" fontSize="10px">
                <Box mr={0}>
                <Tooltip title="Delete">
                <DeleteOutlinedIcon className={classes.iconSpace}/>
                </Tooltip>
                </Box>
              </Link>
             </Box>
          </Grid>
        </Box>
        <Grid item xs={12}><Divider /></Grid>
      </Grid>
    );
  });
  return (
    <Paper>
      <Hidden xsDown>
        <Grid key={0} container>
          <Box flexGrow={1} display="flex" >
            <Grid item xs={3} >
              <Box textAlign="left" my={2} pl={3}><FormattedMessage id="Label.Product" defualtMessage="Product" /></Box>
            </Grid>
            <Grid item xs={3}>
              <Box textAlign="left" my={2} xs={3} ><FormattedMessage id="Label.Description" defualtMessage="Description" /></Box>
            </Grid>
            <Grid item xs={2}>
              <Box textAlign="left" my={2} xs={3}><FormattedMessage id="Label.Quantity" defualtMessage="Quantity" /></Box>
            </Grid>
          </Box>
          <Grid item xs={2}>
            <Box textAlign="right" my={2} pr={3} xs={3}><FormattedMessage id="Label.Price" defualtMessage="Price" /></Box>
          </Grid>
          <Grid item xs={12}><Divider /></Grid>
        </Grid>
        {pcView}
      </Hidden>
      <Hidden smUp>
        {mobileView}
      </Hidden>
      <Netprice />
    </Paper>
  );
};
export default CheckoutOrder;
