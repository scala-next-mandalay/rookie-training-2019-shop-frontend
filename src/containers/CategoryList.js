import { connect } from 'react-redux';
import CategoryList from '../components/CategoryList';
import { setCategoryId } from '../modules/items';

export default connect(
  (state) => ({
    categories: state.categories.rows,
  }),
  (dispatch) => ({
    setCategoryId: (id) => dispatch(setCategoryId(id)),
  })
)(CategoryList);
