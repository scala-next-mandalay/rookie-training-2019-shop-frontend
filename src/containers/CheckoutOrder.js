import { connect } from 'react-redux';
import CheckoutOrder from '../components/CheckoutOrder';
import { deleteCartItem ,changeQuantity} from '../modules/cart';


export default connect(
  (state) => ({
    cart: state.cart.rows,
    totalPrice: state.cart.totalPrice,
    totalQuantity: state.cart.totalQuantity,
    maxQuantity: state.cart.maxQuantity
  }),
  (dispatch) => ({
    deleteCartItem: (itemId) => dispatch(deleteCartItem(itemId)),
    changeQuantity: (itemId, quantity) => dispatch(changeQuantity(itemId, quantity)),    

  })
)(CheckoutOrder);