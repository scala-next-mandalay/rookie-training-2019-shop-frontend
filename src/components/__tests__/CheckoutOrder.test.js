
import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import CategoryList from '../CategoryList';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import CheckoutOrder from '../CheckoutOrder';
/*global jest*/
/*global expect*/
/*global MouseEvent*/
/*global Event*/

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
  
window.innerWidth = 700;
window.innerHeight = 700;
window.dispatchEvent(new Event('resize'));
})



afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null

})



const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;

describe("CheckoutOrder component", () => {
  let cart = [{ id: 1, name: "a", quantity: 1, price: 1000, image:"a.jpg",subTotal : 1000 },
             { id: 2, name: "b", quantity: 1, price: 1000 , image:"b.jpg",subTotal : 1000 }];
 const deleteCartItem = jest.fn((id) => {

      if (id === 1) cart = [{ id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }];

      else cart = [{ id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 }];

    });
  it('matches the snapshot', () => {
    const CheckoutOrderSnapshot = renderer.create(
    <Parent>
    <CheckoutOrder
      cart={cart}
      totalPrice={2000}
      totalQuantity={2}
      deleteCartItem={deleteCartItem}
    /></Parent>).toJSON();
    expect(CheckoutOrderSnapshot).toMatchSnapshot();

  });

});


describe("testing cart table", () => {

  it('testing delete link in checkoutOrder', () => {
    let cart = [
      { id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 },
      { id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }
    ];
    const deleteCartItem = jest.fn((id) => {
      if (id === 1) cart = [{ id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }];
      else cart = [{ id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 }];
    });
     act(() => {
      ReactDOM.render((
        <Parent><CheckoutOrder 
                    cart={cart} 
                    totalPrice={2000}
                    totalQuantity={2} 
                    deleteCartItem={deleteCartItem} /></Parent>

      ), container);
    });
  const linkArr = document.querySelectorAll('a');
  //console.log("linkArr",linkArr.length)
  // console.log(document.body.outerHTML)

  act(() => {
  linkArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
  });
  const expectedCart =[{ id: 2, name: "b", quantity: 1, price: 1000, subTotal: 1000 }];
  expect(deleteCartItem).toHaveBeenCalled();
  expect(cart).toStrictEqual(expectedCart);

  act(() => {
  linkArr[1].dispatchEvent(new MouseEvent('click', { bubbles: true }));
  });
  const expectedCart1 =[{ id: 1, name: "a", quantity: 1, price: 1000, subTotal: 1000 }];
  expect(deleteCartItem).toHaveBeenCalled();
  expect(cart).toStrictEqual(expectedCart1);
  });
  
  it('testing and checking recieved data in checkoutOrder', () => {

    let cart = [
      { id: 1, name: "a", quantity: 5, price: 500, subTotal: 2500 },
      { id: 2, name: "b", quantity: 3, price: 1000, subTotal: 3000 }
    ];
     act(() => {

      ReactDOM.render((
        <Parent><CheckoutOrder 
                    cart={cart} 
                    totalPrice={2000}
                    totalQuantity={2} 
                    /></Parent>
      ), container);
      const divArr3 = document.querySelectorAll('div span');
    //   console.log("divArr",divArr3.length)
    //   console.log("#1",divArr3[0].textContent)
    // console.log("#2",divArr3[1].textContent)
    // console.log("#3",divArr3[2].textContent)
    // console.log("#4",divArr3[3].textContent)
    // console.log("#5",divArr3[4].textContent)
    // console.log("#6",divArr3[5].textContent)
    // console.log("#7",divArr3[6].textContent)
    // console.log("#8",divArr3[7].textContent)
    // console.log("#9",divArr3[8].textContent)
    // console.log("#10",divArr3[9].textContent)

    });
  });
})
