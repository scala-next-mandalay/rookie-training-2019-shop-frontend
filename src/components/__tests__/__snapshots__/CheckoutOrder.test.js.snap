// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutOrder component matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="PrivateHiddenCss-xsDown-36"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiBox-root MuiBox-root-153"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-154"
          >
            <span>
              Product
            </span>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-155"
            xs={3}
          >
            <span>
              Description
            </span>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-156"
            xs={3}
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-157"
          xs={3}
        >
          <span>
            Price
          </span>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root text2 MuiGrid-container"
    >
      <div
        className="MuiBox-root MuiBox-root-163"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-164"
          >
            <div
              className="MuiCardMedia-root makeStyles-descImagePC-3"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/a.jpg\\")",
                }
              }
              title="a"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-167 nameTxt"
          >
            a
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-168"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <select
                className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={4}
                >
                  4
                </option>
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={7}
                >
                  7
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={9}
                >
                  9
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={11}
                >
                  11
                </option>
                <option
                  value={12}
                >
                  12
                </option>
                <option
                  value={13}
                >
                  13
                </option>
                <option
                  value={14}
                >
                  14
                </option>
                <option
                  value={15}
                >
                  15
                </option>
                <option
                  value={16}
                >
                  16
                </option>
                <option
                  value={17}
                >
                  17
                </option>
                <option
                  value={18}
                >
                  18
                </option>
                <option
                  value={19}
                >
                  19
                </option>
                <option
                  value={20}
                >
                  20
                </option>
              </select>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiNativeSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-215"
        >
          <div
            className="MuiBox-root MuiBox-root-216"
          >
            (
            1
            x
            1000
            )MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-217"
          >
             
            1000
            MMK 
          </div>
          <div
            className="MuiBox-root MuiBox-root-218"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-btnRemoveItem-5 MuiTypography-colorSecondary"
              fontSize="10px"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <div
                className="MuiBox-root MuiBox-root-255"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="presentation"
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root text2 MuiGrid-container"
    >
      <div
        className="MuiBox-root MuiBox-root-264"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-265"
          >
            <div
              className="MuiCardMedia-root makeStyles-descImagePC-3"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/b.jpg\\")",
                }
              }
              title="b"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
        >
          <div
            className="MuiBox-root MuiBox-root-266 nameTxt"
          >
            b
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            className="MuiBox-root MuiBox-root-267"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <select
                className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={4}
                >
                  4
                </option>
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={7}
                >
                  7
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={9}
                >
                  9
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={11}
                >
                  11
                </option>
                <option
                  value={12}
                >
                  12
                </option>
                <option
                  value={13}
                >
                  13
                </option>
                <option
                  value={14}
                >
                  14
                </option>
                <option
                  value={15}
                >
                  15
                </option>
                <option
                  value={16}
                >
                  16
                </option>
                <option
                  value={17}
                >
                  17
                </option>
                <option
                  value={18}
                >
                  18
                </option>
                <option
                  value={19}
                >
                  19
                </option>
                <option
                  value={20}
                >
                  20
                </option>
              </select>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiNativeSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          className="MuiBox-root MuiBox-root-268"
        >
          <div
            className="MuiBox-root MuiBox-root-269"
          >
            (
            1
            x
            1000
            )MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-270"
          >
             
            1000
            MMK 
          </div>
          <div
            className="MuiBox-root MuiBox-root-271"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-btnRemoveItem-5 MuiTypography-colorSecondary"
              fontSize="10px"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <div
                className="MuiBox-root MuiBox-root-272"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="presentation"
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
  </div>
  <div
    className="PrivateHiddenCss-smUp-38"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiBox-root MuiBox-root-273"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiBox-root MuiBox-root-274"
          >
            <div
              className="MuiCardMedia-root makeStyles-descImagePC-3"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/a.jpg\\")",
                }
              }
              title="a"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root text2 MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiBox-root MuiBox-root-275"
          >
            a
          </div>
          <div
            className="MuiBox-root MuiBox-root-276"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <select
                className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={4}
                >
                  4
                </option>
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={7}
                >
                  7
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={9}
                >
                  9
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={11}
                >
                  11
                </option>
                <option
                  value={12}
                >
                  12
                </option>
                <option
                  value={13}
                >
                  13
                </option>
                <option
                  value={14}
                >
                  14
                </option>
                <option
                  value={15}
                >
                  15
                </option>
                <option
                  value={16}
                >
                  16
                </option>
                <option
                  value={17}
                >
                  17
                </option>
                <option
                  value={18}
                >
                  18
                </option>
                <option
                  value={19}
                >
                  19
                </option>
                <option
                  value={20}
                >
                  20
                </option>
              </select>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiNativeSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-277"
          >
            1
            x 
            1000
            MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-278"
          >
            1000
             MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-279"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-btnRemoveItem-5 MuiTypography-colorSecondary"
              fontSize="10px"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <div
                className="MuiBox-root MuiBox-root-280"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden="true"
                  className="MuiSvgIcon-root makeStyles-iconSpace-6"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="presentation"
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiBox-root MuiBox-root-281"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiBox-root MuiBox-root-282"
          >
            <div
              className="MuiCardMedia-root makeStyles-descImagePC-3"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/b.jpg\\")",
                }
              }
              title="b"
            />
          </div>
        </div>
        <div
          className="MuiGrid-root text2 MuiGrid-item MuiGrid-grid-xs-6"
        >
          <div
            className="MuiBox-root MuiBox-root-283"
          >
            b
          </div>
          <div
            className="MuiBox-root MuiBox-root-284"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <select
                className="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={1}
              >
                <option
                  value={1}
                >
                  1
                </option>
                <option
                  value={2}
                >
                  2
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={4}
                >
                  4
                </option>
                <option
                  value={5}
                >
                  5
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={7}
                >
                  7
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={9}
                >
                  9
                </option>
                <option
                  value={10}
                >
                  10
                </option>
                <option
                  value={11}
                >
                  11
                </option>
                <option
                  value={12}
                >
                  12
                </option>
                <option
                  value={13}
                >
                  13
                </option>
                <option
                  value={14}
                >
                  14
                </option>
                <option
                  value={15}
                >
                  15
                </option>
                <option
                  value={16}
                >
                  16
                </option>
                <option
                  value={17}
                >
                  17
                </option>
                <option
                  value={18}
                >
                  18
                </option>
                <option
                  value={19}
                >
                  19
                </option>
                <option
                  value={20}
                >
                  20
                </option>
              </select>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiNativeSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-285"
          >
            1
            x 
            1000
            MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-286"
          >
            1000
             MMK
          </div>
          <div
            className="MuiBox-root MuiBox-root-287"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-btnRemoveItem-5 MuiTypography-colorSecondary"
              fontSize="10px"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <div
                className="MuiBox-root MuiBox-root-288"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden="true"
                  className="MuiSvgIcon-root makeStyles-iconSpace-6"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  role="presentation"
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 21h12V7H6v14zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-289"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-290"
      >
         
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-291"
      >
        <span>
          Total Quantity
        </span>
      </div>
      <div
        className="MuiBox-root MuiBox-root-292"
      >
        <span>
          SubTotal
        </span>
         
      </div>
      <div
        className="MuiBox-root MuiBox-root-293"
      >
        <span>
          Sales Tax
        </span>
      </div>
      <div
        className="MuiBox-root MuiBox-root-294"
      >
        <span>
          Total Price
        </span>
      </div>
    </div>
    <div
      className="MuiGrid-root text1 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4"
    >
      <div
        className="MuiBox-root MuiBox-root-295"
      >
        2
         (Items)
      </div>
      <div
        className="MuiBox-root MuiBox-root-296"
      >
         
        2000
         MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-297"
      >
         0.0 MMK 
      </div>
      <div
        className="MuiBox-root MuiBox-root-298"
      >
         
        2000
         MMK
      </div>
    </div>
  </div>
</div>
`;
